File: app/models/page/sections/baby/milestone.rb
class Milestone
  include Mongoid::Document
  include Mongoid::Timestamps
  include Mongoid::Orderable

  embedded_in :baby_grows_widget

  field :date,   :type => Date
  field :age,    :type => String
  field :weight, :type => String
  field :height, :type => String

  validates_presence_of :date
  validates_presence_of :age
  validates_presence_of :weight
  validates_presence_of :height

  def formatted_date
    self.date ? self.date.stamp('2013-11-30') : ''
  end

  orderable

  default_scope order_by(:date.asc)

end

File: app/assets/javascripts/baby_grows/milestone.js
/* globals ko, Chippd */
(function() {
  "use strict";
  Chippd.Milestone = Chippd.Object.extend({
    constructor: function () {
      var self = this;
      self._id = ko.observable();
      self.date = ko.observable();
      self.age = ko.observable();
      self.weight = ko.observable();
      self.height = ko.observable();
    }
  });
})();
